MoreLinq.Test/RandomTest.cs

```diff
@@ -17,74 +17,86 @@ public class RandomTest
         /// Verify that passing an <c>null</c> random generator results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestNullGeneratorException1()
         {
-            MoreEnumerable.Random(null);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                MoreEnumerable.Random(null);
+            });
         }
 
         /// <summary>
         /// Verify that passing an <c>null</c> random generator results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestNullGeneratorException2()
         {
-            const int maxValue = 10;
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const int maxValue = 10;
 
-            Assert.Greater( maxValue, 0 );
-            MoreEnumerable.Random(null, maxValue);
+                Assert.Greater(maxValue, 0);
+                MoreEnumerable.Random(null, maxValue);
```

Comment:
Only this line should be inside the lambda to Assert.Throws as that's what will cause and being tested for an exception.